crates/core/src/protobuf.rs

```diff
@@ -34,9 +38,133 @@ pub mod node_ctl_svc {
 }
 
 pub mod metadata_proxy_svc {
+    use std::sync::Arc;
+
+    use bytestring::ByteString;
+    use tokio::sync::Mutex;
+    use tonic::{codec::CompressionEncoding, transport::Channel};
+
+    use metadata_proxy_svc_client::MetadataProxySvcClient;
+    use restate_types::{
+        Version,
+        metadata::{Precondition, VersionedValue},
+        nodes_config::NodesConfiguration,
+    };
+
+    use crate::metadata_store::{MetadataStore, ProvisionError, ReadError, WriteError};
+
+    use super::{to_read_err, to_write_err};
 
     tonic::include_proto!("restate.metadata_proxy_svc");
 
     pub const FILE_DESCRIPTOR_SET: &[u8] =
         tonic::include_file_descriptor_set!("metadata_proxy_svc_descriptor");
+
+    pub struct MetadataStoreProxy {
+        client: Arc<Mutex<MetadataProxySvcClient<Channel>>>,
+    }
+
+    impl MetadataStoreProxy {
```

Comment:
Nitpicky but any chance we could move the implementation to its own Rust module? I think it would be much more discoverable; can be in a follow up PR if you plan to do more work in this area.